core/controllers/editor.py

```diff
@@ -701,33 +702,20 @@ def get(self, exploration_id):
                 'Unrecognized output format %s' % output_format)
 
 
-class StateDownloadHandler(EditorHandler):
-    """Downloads a state as a YAML string."""
+class StateYamlHandler(EditorHandler):
+    """Given a representation of a state, converts it to a YAML string."""
 
-    def get(self, exploration_id):
+    def get(self):
         """Handles GET requests."""
-        try:
-            exploration = exp_services.get_exploration_by_id(exploration_id)
-        except:
-            raise self.PageNotFoundException
+        state_dict = json.loads(self.request.get('stringified_state'))
```

Comment:
Hm, I still think that adding a library is a more invasive fix than doing a backend call which isn't too dissimilar to what we were doing before.

I think of this as a bit like reader.classify(), which is just a stateless backend service (though in this case the handler just verifies a json dict and, if it's in a recognized format, converts it to yaml).